src/uct/tcp/tcp_sockcm_ep.c

```diff
@@ -754,15 +807,20 @@ ucs_status_t uct_tcp_sockcm_ep_create(const uct_ep_params_t *params, uct_ep_h *e
         tcp_sockcm = uct_tcp_sockcm_ep_get_cm(tcp_ep);
         UCS_ASYNC_BLOCK(tcp_sockcm->super.iface.worker->async);
 
-        /* the server's endpoint was already created by the listener, return it */
-        *ep_p  = &tcp_ep->super.super.super;
-
         /* fill the tcp_ep fields from the caller's params */
-        status = uct_cm_set_common_data(&tcp_ep->super, params);
+        status = uct_cm_use_ep_params(&tcp_ep->super, params);
```

Comment:
should move UCT_CM_SET_CB in line 825 after this because it's also setting cep field from param

Reply:
this function does the params check so it checks that the UCT_EP_PARAM_FIELD_CM is even set before i can update the ep with the (new) cm from parmas.

Reply:
can we move it to after calling the function?

Reply:
can you please clarify? move what to after what?

before calling `uct_tcp_sockcm_ep_server_update_cm`, need to call `uct_cm_set_common_data` to check that the UCT_EP_PARAM_FIELD_CM is set

Reply:
after uct_cm_set_common_data

Reply:
oh, i thought you meant something else.
so you meant the order to be:

uct_cm_set_common_data
UCT_CM_SET_CB (for notify_cb)
uct_tcp_sockcm_ep_server_update_cm

right?

Reply:
yes